src/client/systems/ClientGameController.cpp

```diff
@@ -48,7 +48,14 @@ InputDef::PlayerInputs ClientGameController::getPlayerInputs() {
 }
 
 void ClientGameController::preUpdate() {
+    // call collectinputs to process window events, but if it's bot mode
+    // then override any user commands
     auto inputs = m_inputController->collectInputs(m_window);
+    if (IS_BOT) {
+        inputs = m_gameObjectController->getBotMove(m_playerId);
+    }
+    sendInputs(inputs);
+    saveInputs(inputs);
```

Comment:
Why do you have these here? 
They are called bellow in line 68-69 while the game status is `playing`. No reason to send/save moves before the game started.

Reply:
nice catch bad copy pasta